### ISSUE TITLE
Using my own reference shoreline geojson file

### ISSUE THREAD

**Issue by pgurruchaga** [kvos/CoastSat#223]
Hi! I have a geojson file with points that I want to use as reference shoreline. I've read issue #94 that adress this but I still haven't been able to solve this.
The shape has points every ten meters (in the right coordinate system as output epsg and with attributes x and y) and it corresponds to the polygon zone that I'm using, so I don't have to specifiy anything regarding to this. I don't know if I need a .pkl file or if geojson is enough, because it doesn't work. So far I've use this:

`ref_shorelines = os.path.join(inputs['filepath'], inputs['sitename'], sitename + '_reference_shoreline.geojson') `
`settings['reference_shoreline'] = np.array(ref_shorelines) `
`settings['max_dist_ref'] = 100 # max distance (in meters) allowed from the reference shoreline `

that seems to work but then I get an error when trying to map shorelines, so I would really appreciate the help.

Thanks a lot,
Paula.

**Comment by kvos** [comment]
hi @pgurruchaga , you need to read the geojson file first with geopandas, then convert the geometries to numpy array.
You can see some examples in my SDS_tools.py module. Let me know if you manage to get it running as it would be a good add-on to the toolbox to be able to load a reference shoreline from geojson

**Comment by pgurruchaga** [comment]
Sorry, couldn't do it with the geojson, I managed to read with geopandas and then convert as you said but I didn't do it right, because it led to errors at the next step mapping the shorelines. 
Anyway, I exported the points to a .csv file and loaded them on CoastSat like this:

`ref_shorelines = os.path.join(inputs['filepath'], inputs['sitename'], sitename + '_reference_shoreline.csv')`
`ref_shorelines=numpy.loadtxt(open(ref_shorelines, "rb"), delimiter=",")`
`settings['reference_shoreline']=ref_shorelines`
`settings['max_dist_ref'] = 100 # max distance (in meters) allowed from the reference shoreline`

I know it's not the best but it worked perfectly, I tested it by mapping the shorelines without any reference at all and obtaining different lines.

Thanks for the help!
